--- i/src/command_pyroscope.cc
+++ w/src/command_pyroscope.cc
@@ -49,7 +49,7 @@
 
 // In 0.9.x this changed to 'tr1' (dropping sigc::bind), see https://stackoverflow.com/a/4682954/2748717
 // "C++ Technical Report 1" was later added to "C++11", using tr1 makes stuff compile on older GCC
-#define _cxxstd_ tr1
+#define _cxxstd_ std
 
 // List of system capabilities for `system.has` command
 static std::set<std::string> system_capabilities;
@@ -60,7 +60,7 @@ int log_messages_fd = -1;
 };
 
 
-#if RT_HEX_VERSION <= 0x000906
+#if RT_HEX_VERSION <= 0x000907
 // will be merged into 0.9.7+ mainline!
 
 namespace torrent {
@@ -1431,8 +1431,10 @@ void initialize_command_pyroscope() {
     CMD2_ANY_STRING("log.messages", _cxxstd_::bind(&cmd_log_messages, _cxxstd_::placeholders::_2));
     CMD2_ANY_P("import.return", &cmd_import_return);
     CMD2_ANY("do", _cxxstd_::bind(&cmd_do, _cxxstd_::placeholders::_1, _cxxstd_::placeholders::_2));
+#if RT_HEX_VERSION <= 0x000906
     CMD2_DL("d.is_meta", _cxxstd_::bind(&torrent::DownloadInfo::is_meta_download,
                                         _cxxstd_::bind(&core::Download::info, _cxxstd_::placeholders::_1)));
+#endif
 
     // List capabilities of this build
     add_capability("system.has");         // self
--- i/src/ui_pyroscope.cc
+++ w/src/ui_pyroscope.cc
@@ -45,7 +45,7 @@ python -c 'print u"\u22c5 \u22c5\u22c5 \u201d \u2019 \u266f \u2622 \u260d \u2318
 
 // In 0.9.x this changed to 'tr1', see https://stackoverflow.com/a/4682954/2748717
 // "C++ Technical Report 1" was later added to "C++11", using tr1 makes stuff compile on older GCC
-#define _cxxstd_ tr1
+#define _cxxstd_ std
 
 #define D_INFO(item) (item->info())
 #include "rpc/object_storage.h"
@@ -898,7 +898,7 @@ bool ui_pyroscope_download_list_redraw(Window* window, display::Canvas* canvas,
         char* last = buffer + canvas->width() + 1;
 
         pos = canvas->height() - 2 - network_history_lines;
-        print_download_info(buffer, last, *view->focus());
+        print_download_info_full(buffer, last, *view->focus());
         canvas->print(3, pos, "%s", buffer);
         canvas->set_attr(0, pos, -1, attr_map[ps::COL_LABEL], ps::COL_LABEL);
         print_download_status(buffer, last, *view->focus());
