diff --git i/src/rpc/command_map.cc w/src/rpc/command_map.cc
index 17f253b..1001d08 100644
--- i/src/rpc/command_map.cc
+++ w/src/rpc/command_map.cc
@@ -77,7 +77,7 @@ CommandMap::insert(key_type key, int flags, const char* parm, const char* doc) {
   iterator itr = base_type::find(key);
 
   if (itr != base_type::end())
-    throw torrent::internal_error("CommandMap::insert(...) tried to insert an already existing key.");
+    throw torrent::internal_error("CommandMap::insert(...) tried to insert an already existing key: " + std::string(key));
 
   // TODO: This is not honoring the public_xmlrpc flags!!!
   if (rpc::xmlrpc.is_valid() && (flags & flag_public_xmlrpc))
